v2/index_gen_test.go

```diff
@@ -153,3 +164,93 @@ func TestGenerateIndexFromFile(t *testing.T) {
 		})
 	}
 }
+
+func TestMultihashIndexSortedConsistencyWithIndexSorted(t *testing.T) {
+	path := "testdata/sample-v1.car"
+
+	sortedIndex, err := carv2.GenerateIndexFromFile(path)
+	require.NoError(t, err)
+	require.Equal(t, multicodec.CarIndexSorted, sortedIndex.Codec())
+
+	f, err := os.Open(path)
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, f.Close()) })
+	br, err := carv2.NewBlockReader(f)
+	require.NoError(t, err)
+
+	subject := generateMultihashSortedIndex(t, path)
+	for {
+		wantNext, err := br.Next()
+		if err == io.EOF {
+			break
+		}
+		require.NoError(t, err)
+
+		dmh, err := multihash.Decode(wantNext.Cid().Hash())
+		require.NoError(t, err)
+		if dmh.Code == multihash.IDENTITY {
+			continue
+		}
+
+		wantCid := wantNext.Cid()
+		var wantOffsets []uint64
+		err = sortedIndex.GetAll(wantCid, func(o uint64) bool {
+			wantOffsets = append(wantOffsets, o)
+			fmt.Println(wantCid)
+			return false
+		})
+		require.NoError(t, err)
+
+		var gotOffsets []uint64
+		err = subject.GetAll(wantCid, func(o uint64) bool {
+			gotOffsets = append(gotOffsets, o)
+			fmt.Println(wantCid)
+			return false
+		})
+		fmt.Println("----------------")
+		require.NoError(t, err)
+		require.Equal(t, wantOffsets, gotOffsets)
+	}
+}
+
+func generateMultihashSortedIndex(t *testing.T, path string) index.Index {
+	f, err := os.Open(path)
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, f.Close()) })
+	reader := internalio.ToByteReadSeeker(f)
+	header, err := carv1.ReadHeader(reader)
+	require.NoError(t, err)
+	require.Equal(t, uint64(1), header.Version)
+
+	idx, err := index.New(index.IndexMultihashSortedCodec)
+	require.NoError(t, err)
+	records := make([]index.Record, 0)
+
+	var sectionOffset int64
+	sectionOffset, err = reader.Seek(0, io.SeekCurrent)
+	require.NoError(t, err)
+
+	for {
+		sectionLen, err := varint.ReadUvarint(reader)
+		if err == io.EOF {
+			break
+		}
+		require.NoError(t, err)
+
+		if sectionLen == 0 {
+			break
+		}
+
+		cidLen, c, err := cid.CidFromReader(reader)
+		require.NoError(t, err)
+		records = append(records, index.Record{Cid: c, Offset: uint64(sectionOffset)})
+		remainingSectionLen := int64(sectionLen) - int64(cidLen)
+		sectionOffset, err = reader.Seek(remainingSectionLen, io.SeekCurrent)
+		require.NoError(t, err)
+	}
+
+	err = idx.Load(records)
```

Comment:
extend to make sure you can serialize the index to disk and load it back again?

Reply:
Resolved [here](https://github.com/ipld/go-car/blob/c28358b0377ab1c4468197a4d0e5a0bb69cb7f0c/v2/index/mhindexsorted_test.go)